src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+  },
+
+  _selectOption(option) {
+    this.requestChange({ $set: option });
+    this._clearFilterValue();
+    this._closeList();
+    this._blurFilterInput();
+  },
+
+  //State Helpers: openGroupsId
+
+  _isGroupOpen(openGroupsId, group) {
+    return _.contains(openGroupsId, group.id);
+  },
+
+  _toggleGroupOpenState(group) {
+    let { openGroupsId } = this.state,
+        isOpened = this._isGroupOpen(openGroupsId, group);
+
+    if (isOpened) {
+      this._closeGroup(group);
+    } else {
+      this._openGroup(group);
+    }
+  },
+
+  _openGroup(group) {
+    let { openGroupsId } = this.state;
+    this.setState({ openGroupsId: update(openGroupsId, {$push: [group.id]}) });
+  },
+
+  _closeGroup(group) {
+    let { openGroupsId } = this.state;
+    let index = _.findIndex(openGroupsId, (id) => id === group.id);
+    this.setState({ openGroupsId: update(openGroupsId, {$splice: [[index, 1]]}) });
+  },
+
+  _openSuggestedGroup(suggestionOption) {
+    if (!suggestionOption) {
+      return;
+    }
+    let { options } = this.props,
+        { openGroupsId } = this.state;
+
+    let suggestedGroup = _.find(options, (group) => group.isGroup && _.contains(_.pluck(group.options, 'id'), suggestionOption.id));
+    if (suggestedGroup && !this._isGroupOpen(openGroupsId, suggestedGroup)) {
+      this._openGroup(suggestedGroup);
+    }
+  },
+
+  _openAllGroups() {
+    let {options} = this.props;
+    let allGroupsIds = _.pluck(_.filter(options, 'isGroup'), 'id');
+    this.setState({ openGroupsId: allGroupsIds });
+  },
+
+  _closeAllGroups() {
+    this.setState({ openGroupsId: [] });
+  },
+
+  //State Helpers: filterValue
+
+  _updateFilterValue(newFilterValue) {
+    this.setState({filterValue: newFilterValue});
+    this._suggestFirstOption(newFilterValue);
+  },
+
+  _clearFilterValue() {
+    this._updateFilterValue('');
+  },
+
+  //State Helpers: suggestedOption
+
+  _suggestFirstOption(filterValue) {
+    let suggestion = this._getOptionsIterator(filterValue)[0];
+    this._setSuggestedOption(suggestion);
+  },
+
+  _suggestPreviousOption(filterValue, suggestedOption) {
+    let optionsIterator = this._getOptionsIterator(filterValue);
+
+    let currentSuggestionIndex = suggestedOption ? _.findIndex(optionsIterator, (option) => option.id === suggestedOption.id) : -1;
+    let previousSuggestionIndex = currentSuggestionIndex > 0 ? currentSuggestionIndex - 1 : currentSuggestionIndex;
+
+    this._setSuggestedOption(optionsIterator[previousSuggestionIndex]);
+  },
+
+  _suggestNextOption(filterValue, suggestedOption) {
+    let optionsIterator = this._getOptionsIterator(filterValue);
+
+    let currentSuggestionIndex = suggestedOption ? _.findIndex(optionsIterator, (option) => option.id === suggestedOption.id) : -1;
+    let nextSuggestionIndex = currentSuggestionIndex < (optionsIterator.length - 1) ? currentSuggestionIndex + 1 : currentSuggestionIndex;
+
+    this._setSuggestedOption(optionsIterator[nextSuggestionIndex]);
+  },
```

Comment:
These two methods are very similar, maybe provide a `_moveSuggestion(n)` instead? You can use `Math.min` and `Math.max` to keep it between two bounds.